src/Nethermind/Nethermind.Serialization.FluentRlp/Rlp.cs

```diff
@@ -37,3 +33,43 @@ public static T Read<T>(ReadOnlySpan<byte> source, RefRlpReaderFunc<T> func)
         return result;
     }
 }
+
+/// <remarks>
+/// The existing <see cref="ArrayBufferWriter{T}"/> performs various bound checks and supports resizing buffers
+/// which we don't need for our use case.
+/// </remarks>
+internal class FixedArrayBufferWriter<T> : IBufferWriter<T>
+{
+    private readonly T[] _buffer;
+    private int _index;
+
+    public T[] Buffer => _buffer;
+
+    /// <summary>
+    /// Creates an instance of an <see cref="FixedArrayBufferWriter{T}"/>, in which data can be written to,
+    /// with a fixed capacity specified.
+    /// </summary>
+    /// <param name="capacity">The capacity of the underlying buffer.</param>
+    public FixedArrayBufferWriter(int capacity)
+    {
+        _buffer = new T[capacity];
```

Comment:
ArrayPool?

Reply:
By default I think we should go with a plain array to match the `ArrayBufferWriter` behavior. At the end of the day, the `RLP` static class is like a "safe default" API.

If we want more control over the buffers we use we can write a custom `IBufferWriter` as you suggested earlier while using `RlpReader` and `RlpWriter` directly.